lib/model-access-wrapper.js

```diff
@@ -606,8 +606,7 @@ class ModelAccessWrapper extends CrispHooks {
 	 */
 	_checkDiffPermissions(permissions, doc, data) {
 		const grant = permissions.getTargetGrant(this.options.permissionsTarget, data);
-		const patch = Update.createFromDiff(doc.data, data);
-		const fields = _.pull(createUpdate(patch).getUpdatedFields(), '_id');
+		const fields = doc._getRootModifiedFields(data);
```

Comment:
Bad for several reasons.

1. This function only exists in unimodel-mongo, which will make it break for all other unimodel backends.  I'm surprised the test cases pass because the tests for this use a fake in-memory unimodel backend.

2. We don't only want root-level fields.  Only checking permissions on root-level fields is a vast reduction in flexibility.  What if we want someone to be able to read `contact.name` but not `contact.email`?  Remember to think through all the consequences of a change before making it.